.pre-commit-config.yaml

```diff
@@ -11,11 +11,6 @@ repos:
     hooks:
     -   id: black
 
--   repo: https://github.com/pycqa/isort
-    rev: 5.12.0
-    hooks:
-      - id: isort
-        name: isort (python)
```

Comment:
Why did you remove isort?

Reply:
Ah yeah I couldn't get it to pass the pre-commit check on the line:
```py
from inspect import Signature, signature
```
I think isort might not be case-sensitive because it kept swapping these - I tried other solutions like the below:
```py
from inspect import Signature
from inspect import signature as sig_func
```
but that still didn't work - I just concluded having isort wasn't worth the trouble

Reply:
I might be that I've missed something though here - or perhaps that isort needs to use the `black` profile

Reply:
The whole point of `isort` is to sort and not for you to manually sort.

Not production repo, not important.

In general, thanks for the work you put into this. Look promising but honestly I don't get what you doing, too many dunder methods. Some comments / docstrings would help me to get a bit more of an idea.
Also document what you want to implement in the code.

Is think you implantation covers currently only how an attribute is read `Intersection[T,S].x` but not how it is set, i.e. `Intersection[T, S].x = int #  valid or not?` can't be determined, right?

Reply:
> In general, thanks for the work you put into this. Look promising but honestly I don't get what you doing, too many dunder methods. Some comments / docstrings would help me to get a bit more of an idea.
Also document what you want to implement in the code.

Sure, I can add some docs to explain it - I appreciate there's quite a bit of magic method craziness in there!

> Is think you implantation covers currently only how an attribute is read Intersection[T,S].x but not how it is set, i.e. Intersection[T, S].x = int #  valid or not? can't be determined, right?

Yeah this is only intended to be used with getattribute - I could add a setattribute override, but I'm not sure what it would mean in the context of the simulator. The idea is that when you get an attribute it gets you the attribute *type* rather than the attribute itself. I can add a nice error message though for if anyone tries to set an attribute